plugins/search-react/src/components/SearchFilter/hooks.test.tsx

```diff
@@ -77,15 +77,15 @@ describe('SearchFilter.hooks', () => {
         },
       );
 
-      await waitForNextUpdate();
+      await act(async () => {});
```

Comment:
Doesn't it make sense to have a `await waitFor(() => {})` here instead of an empty `act()`?


```ts
 await waitFor(() => {
        expect(result.current.filters[expectedFilter]).toEqual(expectedValue);
 });
```

For example

Reply:
Yeah tbh the empty `act`s was a bit of a thing, don't like them too much. It seems better to `render(...); await act(async () => {})` than to `await act(async () => render(...))`. It also is only needed when rendering the component triggers async actions in the background and you want to wait for them to complete, so I feel it's somewhat semantically correct at least. Kinda wish it was possible to just `await act()`.

Not sure about switching it out for `waitFor(...)`, that's a more heavyweight helper and it calls `act()` under the hood. I'm thinking that `act` is what you use when you want for React to settle its state, while `waitFor` is what you use when you want to wait for a specific state, typically from a user point of view such as a button to be visible.

Reply:
Gotcha!